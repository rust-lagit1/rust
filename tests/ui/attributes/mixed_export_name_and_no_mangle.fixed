// issue: rust-lang/rust#47446
//@ run-rustfix
//@ check-pass

#[export_name = "foo"]
pub fn bar() {}

fn main() {}
