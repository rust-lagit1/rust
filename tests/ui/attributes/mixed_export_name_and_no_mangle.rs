// issue: rust-lang/rust#47446
//@ run-rustfix
//@ check-pass

#[no_mangle]
//~^ WARN the `#[no_mangle]` attribute may not be used in combination with `#[export_name]`
#[export_name = "foo"]
pub fn bar() {}

fn main() {}
