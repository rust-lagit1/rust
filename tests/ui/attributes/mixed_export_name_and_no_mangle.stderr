warning: the `#[no_mangle]` attribute may not be used in combination with `#[export_name]`
  --> $DIR/mixed_export_name_and_no_mangle.rs:5:1
   |
LL | #[no_mangle]
   | ^^^^^^^^^^^^ `#[no_mangle]` is ignored
   |
note: `#[export_name = "foo"]` takes precedence
  --> $DIR/mixed_export_name_and_no_mangle.rs:7:1
   |
LL | #[export_name = "foo"]
   | ^^^^^^^^^^^^^^^^^^^^^^
   = note: `#[warn(unused_attributes)]` on by default
help: remove the `#[no_mangle]` attribute
   |
LL - #[no_mangle]
   |

warning: 1 warning emitted

